week2/Task3/Program.cs

```diff
@@ -0,0 +1,72 @@
+﻿var time = new JupiterTime(2,20);
+try
+{
+    var timeIn2Hours = time.AddHours(-3);
+    PrintTime(timeIn2Hours);   
+}
+catch (Exception e)
+{
+    Console.WriteLine(e.Message);
+}
+
+void PrintTime(JupiterTime jt)
+{
+    if(jt.Minutes < 9)
+        Console.WriteLine($"{jt.Hours}:0{jt.Minutes}");
+    else
+        Console.WriteLine($"{jt.Hours}:{jt.Minutes}");
+}
+public class JupiterTime
+{   
+    private int _hours, _minutes;
+
+    public JupiterTime( int hours, int minutes)    
+    { 
+        Hours = hours;
+        Minutes = minutes;
+    }
+    public int Hours { 
+        get
+        {
+            return _hours;
+         } 
+        set
+        {
+            if(value < 0) throw new Exception("hours can not be negative");
+            _hours = value;
+        } 
+    }
+    public int Minutes 
+    { 
+        get
+        {
+            return _minutes;
+         } 
+        set
+        {
+            if(value < 0) throw new Exception("Invaid minutes");
+            else if (value > 59)
+            {
+                _minutes = value % 60;
+                _hours = _hours + value / 60;
+                if( _hours > 10 )
+                {
+                    _hours = _hours % 10;
+                }
+            }
+            else
+                _minutes = value;
+        } 
+    }
+
+    public JupiterTime AddHours(int num)
+    {
+        if( num > 0 && num + _hours > 10) // BONUS for larger number like 11
+            _hours = (num + _hours) % 10 ;
+        else if (num >= 0)
+            _hours = _hours + num;
+        else                              // Bonus for negative numbers
+            _hours = 10 + _hours + num % 10;
```

Comment:
multiplication and division is done before adding and subtracting of the numbers.

In this case if you have:
```
var time = new JupiterTime(2, 20);
var timeIn2Hours = time.AddHours(-2);
PrintTime(timeIn2Hours);
```
Output would be 10:20 and we only have 10 hours a day.

To fix the issue, enclose adding operation inside of the parenthesis:  `_hours = (10 + _hours + num) % 10;`